020-LinuxFundamentals/Student/resources/commands.md

```diff
@@ -0,0 +1,250 @@
+| `wc` | Show the number of words, lines, and bytes in a file using wc | 
+| `ls \| xargs wc` | List number of lines/words/characters in each file in a directory with the xargs command | 
+| `cut -d[delimiter] [filename]` | Cut a section of a file and print the result to standard output | 
+| `[data] \| cut -d[delimiter]` | Cut a section of piped data and print the result to standard output | 
+| `awk '[pattern] {print $0}' [filename]` | Print all lines matching a pattern in a file |
+| `diff [file1] [file2]` | Compare two files and display differences| 
+| `source [filename]` | Read and execute the file content in the current shell| 
+| `[command] \| tee [filename] >/dev/null` | Store the command output in a file and skip the terminal output | 
+
+## Searching
+
+| Command | Description |
+|--------------|--------------|
+| `grep [pattern] [file_name]` | Search for a specific pattern in a file with grep |
+| `grep -r [pattern] [directory_name]` | Recursively search for a pattern in a directory:|
+| `locate [name]` | Find all files and directories related to a particular name: |
+| `find [/folder/location] -name [a]` | List names that begin with a specified character [a] in a specified location [/folder/location] by using the find command: |
+| `find [/folder/location] -size [+100M]` | See files larger than a specified size [+100M] in a folder: |
+
+## Directory Navigation
+
+| Command | Description |
+|--------------|--------------|
+|`tar cf [compressed_file.tar] [file_name]`|Archive an existing file|
+|`tar xf [compressed_file.tar]`|Extract an archived file|
+|`tar czf [compressed_file.tar.gz]`|Create a gzip compressed tar file by running|
+|`gzip [file_name]`  | Compress a file with the .gz extension | 
+
+
+## File Transfer
+
+| Command | Description |
+|--------------|--------------|
+|`scp [file_name.txt] [server/tmp]`|Copy a file to a server directory securely using the Linux scp command|
+|`rsync -a [/your/directory] [/backup/] `|Synchronize the contents of a directory with a backup directory using the rsync command|
+
+## Users and Groups
+
+| Command | Description |
+|--------------|--------------|
+|`id`|See details about the active users|
+|`last`|Show last system logins|
+|`who`|Display who is currently logged into the system with the who command|
+|`w`  | Show which users are logged in and their activity | 
+| `groupadd [group_name]` | Add a new group by typing |
+| `adduser [user_name]` | Add a new user|
+| `usermod -aG [group_name] [user_name]` | Add a user to a group |
+| `sudo [command_to_be_executed_as_superuser]`| Temporarily elevate user privileges to superuser or root using the sudo command | 
+| `userdel [user_name] ` | Delete a user | 
+| `usermod`| Modify user information with |
+| `chgrp [group-name] [directory-name]` | Change directory group|
+
+## Package Installation
+
+| Command | Description |
+|--------------|--------------|
+|`yum list installed`|List all installed packages with yum|
```

Comment:
These yum commands won't work on Ubuntu. Should they be here?

Reply:
I have added this message to the top:

_Please note that those commands are for Linux in general, both RedHat-like and Debian-like distributions. There are some commands in the Package management session which may be specific for Redhat-based such as yum and rpm which do not work in Ubuntu (Debian-based)._